crates/sui-adapter/src/execution_engine.rs

```diff
@@ -75,7 +77,47 @@ pub fn execute_transaction_to_effects<
     TransactionEffects,
     Result<Mode::ExecutionResults, ExecutionError>,
 ) {
-    let mut tx_ctx = TxContext::new(&transaction_signer, &transaction_digest, epoch_data);
+    // Separating out impl so we can call this from other context
+    execute_transaction_to_effects_impl::<Mode, S>(
+        shared_object_refs,
+        temporary_store,
+        transaction_kind,
+        transaction_signer,
+        gas,
+        transaction_digest,
+        transaction_dependencies,
+        move_vm,
+        gas_status,
+        &epoch_data.epoch_id(),
+        epoch_data.epoch_start_timestamp(),
+        protocol_config,
+        enable_expensive_checks
+    )
+}
+
+/// Separating out impl so we can call this from other context
+pub fn execute_transaction_to_effects_impl<
```

Comment:
What's the difference between these two functions? I imagine it's for replay reason?

Reply:
`execute_transaction_to_effects` requires having epoch data structure constructed, but it only uses two fields from the sturct. So I made it a wrapper around `execute_transaction_to_effects_impl` which lets me use `execute_transaction_to_effects_impl` without creating epoch data structures, which are difficult to mock